Automattic/jetpack
# Boost/cleanup json types

The data-sync package implemented a `JSONObject` type, which does not actually map to a JSON object - it accepted and JSON-friendly type (including string, boolean, etc). That causes compile errors / ugly warts in various bits of TypeScript code.

This little friendly PR cleans that up, JSONObject once again refers specifically to an _Object_, and JSONValue is any JSON friendly value.

## Proposed changes:
* Internal cleanup of JSON types for Boost + data sync.

### Other information:

## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:
* Check for TypeScript errors.
* Ensure the Boost plugin can generate Critical CSS (as that serves as a solid API test)
